### PR TITLE
Rework all tests to use pytest

### PR DESCRIPTION
**Pull request by benoit-pierre**
Note:
* the use of `lambda` to capture the line of each definition when parametrizing a test (so it's easier to locate a failing test)
* the way key combo tests are handled can seem a little complicated, but this is in prevision of updating the tests when support for locking keys is added

### CODE REVIEW

**Comment by benoit-pierre** [comment]
* fix engine tests
* fix configuration tests random failures                                                                                                                                             
* fix `Config.__setitem__` implementation
* only create strokes/translations log when needed

File: test/__init__.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
```
Comment (morinted) [inline test/__init__.py]: I think the "tests: add parameterize helper" commit should go first, before it's used in other commits.

Comment (benoit-pierre) [reply test/__init__.py]: It does, but Github reorder commits by date... ¯\\_(ツ)_/¯

Comment (morinted) [reply test/__init__.py]: Oh, that's arbitrary, all right… guess it's all good then

**Pull request merged**